api/api.go

```diff
@@ -0,0 +1,121 @@
+package api
+
+import (
+	"errors"
+	"io"
+	"syscall"
+	"time"
+
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+var (
+	ErrNotImplemented = errors.New("command not implemented")
+	ErrEmptyID        = errors.New("container id cannot be empty")
+)
+
+type ContainerOperations interface {
+	Create(id string, opts CreateOpts) (*CreateResult, error)
+	Delete(id string, opts DeleteOpts) error
+	Kill(id string, sig syscall.Signal, opts KillOpts) error
+	List() ([]Container, error)
+	PS(id string) ([]int, error)
+	Pause(id string) error
+	Resume(id string) error
+	Run(id string, opts CreateOpts) (*CreateResult, error)
+	Start(id string) error
+	State(id string) (*Container, error)
+	Exec(id string, opts ExecOpts) (*CreateResult, error)
```

Comment:
I think `Run` and `Exec` returns CreateResult is not very intuitive, probably `CommandResult` is a better name.